### PR TITLE
[claim_from_delivery]: Migrated into new api.

### PR DESCRIPTION
**Pull request by rgo-odoo**
Task: https://www.odoo.com/web#id=12135&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.vWEJtg6p9ynxJXvK

### CODE REVIEW

File: addons/claim_from_delivery/__openerp__.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
+# -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
+
```
Comment (rim-odoo) [inline addons/claim_from_delivery/__openerp__.py]: unnecessary blank line

File: addons/claim_from_delivery/__openerp__.py [hunk 2]
```diff
@@ -12,8 +13,7 @@
 Adds a Claim link to the delivery order.
 """,
     'data' : [
-              'claim_delivery_view.xml',
-              'claim_delivery_data.xml',],
-    'auto_install': False,
+              'views/claim_delivery_view.xml',
+              'data/claim_delivery_data.xml',],
     'installable': True,
 }
```
Comment (rim-odoo) [inline addons/claim_from_delivery/__openerp__.py]: lint that file

File: addons/claim_from_delivery/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    claim_count_out = fields.Integer(compute='_compute_claim_count_out', string='Claims')
+
+    @api.multi
+    def _compute_claim_count_out(self):
+        for picking in self:
+            data = self.env['crm.claim'].read_group([('ref', '=', ('stock.picking,' + str(picking.id)))], ['ref'], ['ref'])
```
Comment (rim-odoo) [inline addons/claim_from_delivery/models/stock_picking.py]: What a bad idea... a read_group in a loop...

Read_group is useful when you have more than one object to group on. Here, keeping the search_count would have been ok

File: addons/claim_from_delivery/__openerp__.py [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
 
 Adds a Claim link to the delivery order.
 """,
-    'data' : [
-              'claim_delivery_view.xml',
-              'claim_delivery_data.xml',],
-    'auto_install': False,
+    'data': [
+        'views/claim_delivery_view.xml',
+        'data/claim_delivery_data.xml', ],
```
Comment (WHPODOO) [inline addons/claim_from_delivery/__openerp__.py]: You must do like : 
```
'data/claim_delivery_data.xml',
 'views/claim_delivery_view.xml'
```

File: addons/claim_from_delivery/models/stock_picking.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+    claim_count_out = fields.Integer(compute='_compute_claim_count_out', string='Claims')
+
+    @api.multi
+    def _compute_claim_count_out(self):
+        refs = [('stock.picking,' + str(picking_id)) for picking_id in self.ids]
+        claim_data = self.env['crm.claim'].read_group([('ref', 'in', refs)], ['ref'], ['ref'])
+        claim_dict = {int(data['ref'].split(',')[1]): data['ref_count'] for data in claim_data}
+        for data in self:
+            data.claim_count_out = claim_dict.get(data.id)
```
Comment (WHPODOO) [inline addons/claim_from_delivery/models/stock_picking.py]: In this method 3 times ***loop*** and 1 ***read_group*** use try to convert in optimize it.

**Comment by hmo-odoo** [comment]
claim_from_delivery is not exits now. so I close PR

**Pull request closed without merging**